blocks/library/list/index.js

```diff
@@ -118,8 +118,9 @@ registerBlockType( 'core/list', {
 				transform: ( { value, citation } ) => {
 					return createBlock( 'core/list', {
 						nodeName: 'UL',
-						values: value.map( ( item, index ) => <li key={ index } >{ get( item, 'children.props.children', '' ) } </li> )
-							.concat( citation ? <li key="citation">{ citation }</li> : [] ),
+						values: value.map( item => get( item, 'children.props.children' ) ).concat( citation )
+							.filter( text => typeof text === 'string' )
```

Comment:
This was a nice find that we are creating a list with an empty <li> and this affects some functionality 👍
Unfortunately, we cannot be that strict and force the text to be a string. For example, if we have a quote with items in bold they are ignored in the transformation because these items are not a string.
Also if the user adds a quote with a line then 3 empty lines and another line, I'm not certain if we should just transform to a list with two items or to a list with one item 3 empty items and one last item.
Maybe we can just add a special case that verifies if value and citation are both empty/unset and if yes we just return with a list with no items. e.g: `if ( ( ! value || ! value.length ) && ! citation ) {`.

Reply:
I've fiddled with the logic in e2257f5dc42dd5c34c1bcce99d255f5a7fecb187. There's a surprising amount of edge cases in this logic. 

The logic I opted for is:

* `'Quote' / 'Citation'` → `<li>Quote</li><li>Citation</li>`
* `'' / 'Citation'` → `<li></li><li>Citation</li>`
* `'Quote' / ''` → `<li>Quote</li>`
* `'' / ''` → ``